deployment/k8s-deploy.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+# Set default context / delete / image / target
+export CONTEXT=minikube
+export DELETE=false
+export IMAGE_FULL_VALUE=
+export TARGET=local
+
+# Track numnber of arguments supplied
+export NUMARGS=$#
+
+# Set variables for help
+SCRIPT=`basename ${BASH_SOURCE[0]}`
+NORM=`tput sgr0`
+BOLD=`tput bold`
+REV=`tput smso`
+
+function HELP {
+  echo -e \\n"Help documentation for ${BOLD}${SCRIPT}.${NORM}"\\n
+  echo -e "${REV}Basic usage:${NORM} ${BOLD}$SCRIPT -i image${NORM}"\\n
+  echo "The following optional parameters are recognised:"
+  echo "${REV}-c${NORM}  --Sets the ${BOLD}context${NORM}. The default is ${BOLD}${CONTEXT}${NORM}."
+  echo "${REV}-d${NORM}  --Sets the ${BOLD}delete${NORM} flag. The default is ${BOLD}${DELETE}${NORM}."
+  echo "${REV}-t${NORM}  --Sets the ${BOLD}target${NORM}. The default is ${BOLD}${TARGET}${NORM}."
+  echo -e "${REV}-h${NORM}  --Displays this help message. No further functions are performed."\\n
+  echo -e "Example: ${BOLD}$SCRIPT -i dockerhub/myImage:version -d -c myCluster -t acceptance${NORM}"\\n
+  exit 1
+}
+
+if [ $NUMARGS -eq 0 ]; then
+  HELP
+fi
+
+# Check for missing parameter values
+while getopts "c:dhi:t:" opt; do
```

Comment:
Two while looks perhaps could have been combined in one.
But I'll leave that for you to decide.
Looks good otherwise.